Data Structures/Linked-Lists/RevresingLinkedListTechniques.cpp

```diff
@@ -0,0 +1,238 @@
+// Name: Anuj Goyal	
+// Email-id: anujg935@gmail.com
+
+// This program describes the 4 ways of revesing the linkedList
+//3 are the recursive ways and 1 is iterative way
+
+#include<iostream>
+using namespace std;
+
+
+//Node class defines the structure of linkedList
+class Node{
+	public:
+		int data;
+		Node * next;
+		Node(int data){
+			this->data = data;
+			this->next=NULL;
+		}
+};
+
+
+//take inputs from the user until user enters -1
+//returns the head of the linkedList
+Node* takeInput(){
+	int data;
+	cin>>data;

```

Comment:
sir, if I make it do while loop then it will stop the loop after making -1 as a node to the loop which is not correct. and I think that's correct way because some new beginner students find's it hard to think about a do-while loop.

Reply:
@Anujg935 Thankyou for the explanation.